robot-server/robot_server/service/dependencies.py

```diff
@@ -86,3 +91,42 @@ def get_session_manager(
             motion_lock=motion_lock,
             protocol_manager=protocol_manager)
     return _session_manager_inst
+
+
+async def check_version_header(
+        request: Request,
+        opentrons_version: str = Header(
+            ...,
+            description=f"The requested HTTP API version which must be "
+                        f"'{constants.MIN_API_VERSION}' or higher. To use the "
+                        f"latest version specify "
+                        f"'{constants.API_VERSION_LATEST}'.")
+) -> None:
+    """Dependency that will check that Opentrons-Version header meets
+    requirements."""
+    # Get the maximum version accepted by client
+    header_value = opentrons_version
+    requested_version = (
+        int(opentrons_version)
+        if header_value != constants.API_VERSION_LATEST else
+        constants.API_VERSION
```

Comment:
I think we use `header_value` and `opentrons_version` interchangeably here? For readability, we should just have one or the other.

```suggestion
    requested_version = (
        int(opentrons_version)
        if opentrons_version != constants.API_VERSION_LATEST else
        constants.API_VERSION
```

or

```suggestion
    header_value = opentrons_version
    requested_version = (
        int(header_value)
        if header_value != constants.API_VERSION_LATEST else
        constants.API_VERSION
```

Reply:
Done